### PR TITLE
fix: script hoist behavior

### PR DESCRIPTION
**Pull request by natemoo-re**
## Changes

See https://github.com/snowpackjs/astro/pull/1743

- When a component only included `<script hoist>`, we would print malformed JS because the `doc` was empty when we tried to print it.
- Now, if we hoist out all the contents, we inject an empty `FrontmatterNode` with an empty `TextNode` child.
- This allows the `printToJs` function to keep doing it's thing and returns valid JS.

## Testing

This is not tested because it is a difficult edge case to test for with our current setup

## Docs

Bug fix

### CODE REVIEW

File: internal/printer/print-to-js.go [hunk 1]
```diff
@@ -105,6 +105,7 @@ func render1(p *printer, n *Node, opts RenderOptions) {
 
 	// Render frontmatter (will be the first node, if it exists)
 	if n.Type == FrontmatterNode {
+		fmt.Println("FrontmatterNode")
```
Comment (matthewp) [inline internal/printer/print-to-js.go:108]: Woops

File: internal/printer/print-to-js.go [hunk 2]
```diff
@@ -113,9 +114,13 @@ func render1(p *printer, n *Node, opts RenderOptions) {
 				// `renderBodyStart` will be the index where we should split the frontmatter.
 				// If we don't encounter any of those, `renderBodyStart` will be `-1`
 				renderBodyStart := js_scanner.FindRenderBody([]byte(c.Data))
-				p.addSourceMapping(n.Loc[0])
+				if len(n.Loc) > 0 {
+					p.addSourceMapping(n.Loc[0])
```
Comment (drwpow) [inline internal/printer/print-to-js.go:118]: Nice!

**Review by drwpow — APPROVED** [review]
LGTM!
